### PR TITLE
sqlx-cli: 0.6.2 -> 0.7.0

### PR DESCRIPTION
**Pull request by xrelkd**
###### Description of changes

- Update `sqlx-cli` to 0.7.0 [CHANGELOG](https://github.com/launchbadge/sqlx/blob/main/CHANGELOG.md#070---2023-06-30)
- Add myself as maintainer of `sqlx-cli`

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by aidalgol** [comment]
It appears that the 0.6.x release is still supported and maintained upstream.  Should we keep both versions in nixpkgs?

**Comment by xrelkd** [comment]
Hi @aidalgol, thank for the review.
I am not sure if the upstream will keep supporting and maintaining the `0.6.x` version.

**Comment by xrelkd** [comment]
Hi @aidalgol 

After reviewing the [tags](https://github.com/launchbadge/sqlx/tags) of `sqlx`, I found that the upstream does not keep maintaining and supporting older version like `0.5.x`, `0.6.x`. 

It seems that the upstream only maintains and supports the latest minor version.

**Comment by xrelkd** [comment]
Hi @marsam
Please help us review this PR.
Thank you so much!

**Review by aidalgol — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/rust/sqlx-cli/default.nix [hunk 1]
```diff
@@ -15,25 +14,16 @@
...
-    # https://github.com/launchbadge/sqlx/pull/2228
-    (fetchpatch {
-      name = "fix-rust-1.65-compile.patch";
-      url = "https://github.com/launchbadge/sqlx/commit/2fdf85b212332647dc4ac47e087df946151feedf.patch";
-      hash = "sha256-5BCuIwmECe9qQrdYll7T+UOGwuTBolWEhKNE7GcZqJw=";
-    })
-  ];
-
-  cargoSha256 = "sha256-AbA8L7rkyZfKW0vvjyrcW5eU6jGD+zAqIcEUOJmeqJs=";
+  cargosHash = "sha256-9vbrehtfw6ctIF7uXZPvODx3kkxz+m9h2Uv+2t45I0w=";
```
[the remark this answers is not in the document]
Comment (xrelkd) [reply pkgs/development/tools/rust/sqlx-cli/default.nix]: Hi @aidalgol
Thank you for the review.

**Review by marsam — APPROVED** [review]

**Comment by xrelkd** [comment]
@marsam Thank you!

**Pull request merged**